### PR TITLE
Inject a Redis Connection Manager to allow pooling of ConnectionMultiplexers

### PR DESCRIPTION
**Pull request by adamhathcock**
This is similar to https://github.com/dotnet/extensions/pull/3140 but takes it a step further to not cache the connection in the RedisCache object to allow usage of a multiplexer pool as described here: https://imperugo.gitbook.io/stackexchange-redis-extensions/configuration

Addresses #718

### CODE REVIEW

**Comment by Pilchie** [comment]
@Tratcher or @JunTaoLuo.

**Comment by Tratcher** [comment]
I've asked @BrennanConroy to take a look.

File: src/Caching/StackExchangeRedis/src/RedisCache.cs [hunk 1]
```diff
@@ -11,7 +11,7 @@
 
 namespace Microsoft.Extensions.Caching.StackExchangeRedis
 {
-    public class RedisCache : IDistributedCache, IDisposable
+    public class RedisCache : IDistributedCache
```
Comment (eerhardt) [inline src/Caching/StackExchangeRedis/src/RedisCache.cs:14]: @Tratcher @BrennanConroy - Are we concerned about API breaking changes in this library?

**Comment by adamhathcock** [comment]
Updated for the CheckDisposed change from upstream.

The RedisCache object shouldn't be using publically so I didn't think removing the IDisposable interface should be breaking.  Strictly, it is breaking.  I can add it back.

File: src/Caching/StackExchangeRedis/src/RedisCache.cs (hunk 1 above)
Comment (Tratcher) [reply src/Caching/StackExchangeRedis/src/RedisCache.cs:14]: Yes, we should avoid any API breaks. In this case it's easier to leave an empty Dispose method.

**Comment by adamhathcock** [comment]
Added IDisposable back to RedisCache

**Comment by JunTaoLuo** [comment]
FYI, as part of https://github.com/dotnet/aspnetcore-internal/issues/3707, we are moving this component to https://github.com/dotnet/aspnetcore, please re-open this PR in that repo.

**Pull request closed without merging**